### PR TITLE
FALCON-2071 : Falcon Spark SQL failing with Yarn Client Mode

### PR DESCRIPTION
**Pull request by peeyushb**

### CODE REVIEW

File: oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java [hunk 1]
```diff
@@ -94,21 +92,35 @@ protected ACTION getUserAction(Cluster cluster, Path buildPath) throws FalconExc
         addOutputFeedsAsArgument(argList, cluster);
         addInputFeedsAsArgument(argList, cluster);
 
-        sparkAction.setJar(addUri(sparkFilePath, cluster));
-
-        setSparkLibFileToWorkflowLib(sparkFilePath, entity);
+        // In Oozie spark action, value for jar is either Java jar file path or Python file path.
+        validateSparkJarFilePath(sparkJarFilePath);
+        sparkAction.setJar(getSparkJarFileName(sparkJarFilePath));
+        setSparkLibFileToWorkflowLib(sparkJarFilePath, entity);
```
Comment (PraveenAdlakha) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: just pass sparkJarFilePath.toString less number of changes will be required in the code.

File: oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java [hunk 2]
```diff
@@ -94,21 +92,35 @@ protected ACTION getUserAction(Cluster cluster, Path buildPath) throws FalconExc
...
         }
     }
 
+    private void validateSparkJarFilePath(Path sparkJarFilePath) throws FalconException {
+        if (!sparkJarFilePath.isAbsolute()) {
+            throw new FalconException("Spark jar file path must be absolute:"+sparkJarFilePath);
+        }
+    }
+
+    private String getSparkJarFileName(Path sparkJarFilePath) {
```
Comment (PraveenAdlakha) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: this method is not really required only one call is there and we donot have any logic implemented in it.

File: oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java (hunk 1 above)
Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: Fixed.

File: oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java (hunk 2 above)
Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: Fixed.

**Comment by PraveenAdlakha** [comment]
+ 1 LGTM

**Comment by bvellanki** [comment]
+1 will commit shortly

**Comment by vrangan** [comment]
+1

Also, we can put it in 0.10 as we have validated internally

**Pull request closed without merging**